### ISSUE TITLE
Color coding menus

### ISSUE THREAD

**Issue by Nicko2102** [absoIute/Mega-Hack-Pro-Future#463]
I got this idea after reading @Br1ttank88's comment on #450, when he mentioned he sometimes had trouble finding the search bar, and i noticed that this has happened to me as well. So, even thought it is quite similar to #70, i think it would be really great if we could change the accent colour of each menu separately (e.g. cosmetic to yellow, status to blue, display to red, etc.). It could help with user experience as well as looking pretty cool.

**Comment by Brittank88** [comment]
Feel free to mention me, I don't mind. I didn't like succmydecc mentioning us because it was clear it was just to get our attention """faster""" and that felt rude.

You guys have proven you actually care though, so feel free, at least for me. ❤️

**Comment by MainConMan** [comment]
Also, if this gets implemented, it would be cool to let extension devs to choose a default color for their extensions.

**Comment by Brittank88** [comment]
Had a very fun idea - what if you could just start typing whilst not clicked into any input box, and it draws a line from the cursor to every menu entry (or module if you prefix with `m:`) that matches? It would kinda look like some features a hacked MC client has, but it would look awesome!

I should make concept art of it, to show how I'd account for items that need to be scrolled to.

**Comment by Brittank88** [comment]
This is out-of-scope of this issue specifically but I'll note it before I forget: What about following what GitHub does and having certain flags for your searches? We've discussed `m:` before, which highlights that entire module if it is the only entry in the search, but what about something like... `m:Player invis` and it searches only within the `Player` module for hacks with the substring `invis`?

**Comment by Brittank88** [comment]
I might do what I've done with other related issue sets and roll it into a mega issue.

**Comment by MainConMan** [comment]
> Had a very fun idea - what if you could just start typing whilst not clicked into any input box, and it draws a line from the cursor to every menu entry (or module if you prefix with `m:`) that matches? It would kinda look like some features a hacked MC client has, but it would look awesome!
> 
> I should make concept art of it, to show how I'd account for items that need to be scrolled to.

That sounds very cool, but also sounds like it would be hard to implement, I may be wrong though.

**Comment by Brittank88** [comment]
> > Had a very fun idea - what if you could just start typing whilst not clicked into any input box, and it draws a line from the cursor to every menu entry (or module if you prefix with `m:`) that matches? It would kinda look like some features a hacked MC client has, but it would look awesome!
> > I should make concept art of it, to show how I'd account for items that need to be scrolled to.
> 
> That sounds very cool, but also sounds like it would be hard to implement, I may be wrong though.

https://stackoverflow.com/questions/14486291/how-to-draw-line-in-opengl

**Comment by absoIute** [comment]
I feel like that wouldn't look too pretty but instant-selecting the search bar is definitely something I can add!

**Comment by absoIute** [comment]
As for changing individual modules' colour it's something I may look into with theming but might leave until until a later update like v8 since having each major Mega Hack version easily recognisable is something I'm going for

**Comment by Brittank88** [comment]
> As for changing individual modules' colour it's something I may look into with theming but might leave until until a later update like v8 since having each major Mega Hack version easily recognisable is something I'm going for

For v8 do you plan to get your OGL implementation to be full-capability? Like, I remember in the past you mentioned it was limited.

**Comment by absoIute** [comment]
I'm considering rebuilding it on-top of cocos2d-x so it would have a strong base to begin with, OpenGL has had a few issues especially on AMD/ATI graphics devices

**Comment by Brittank88** [comment]
> I'm considering rebuilding it on-top of cocos2d-x so it would have a strong base to begin with, OpenGL has had a few issues especially on AMD/ATI graphics devices

There's also [Diligent Engine](https://github.com/DiligentGraphics/DiligentEngine) maybe??

**Comment by absoIute** [comment]
Yeah something like that'd work, using cocos2d-x it could limit it a little with scene switching & make porting slower with 2.2 on the "horizon" (even though I very much doubt it)
